From 816be71fe1cbd3944a6144e189f3f15fda38b0c9 Mon Sep 17 00:00:00 2001
From: Mihails Strasuns <mihails.strasuns.contractor@sociomantic.com>
Date: Mon, 4 Jul 2016 15:06:48 +0300
Subject: [PATCH 09/17] Fix undefined behaviour of checking stomping flag

Calling `strcmp` on null pointer (when env var is not present) result
in undefined behaviour.
---
 src/rt/lifetime.d | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/rt/lifetime.d b/src/rt/lifetime.d
index 8d1350f0..665a937e 100644
--- a/src/rt/lifetime.d
+++ b/src/rt/lifetime.d
@@ -251,7 +251,7 @@ void stomping_prevention_trigger_nonpure ( ) nothrow
     else
     {
         flag = getenv("ALLOW_STOMPING_PREVENTION".ptr);
-        if (strcmp(flag, "0") == 0)
+        if (flag && strcmp(flag, "0") == 0)
             assert(false, "Stomping prevention has been triggerred");
     }
 }
-- 
2.11.0

